packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -511,63 +359,190 @@ test.skip('Can proceed through checkout as registered customer', async () => {
     global.server.use(
+        // mock adding guest email to basket
+        rest.put('*/baskets/:basketId/customer', (req, res, ctx) => {
+            currentBasket.customerInfo.email = 'customer@test.com'
+            return res(ctx.json(currentBasket))
+        }),
+
         // mock fetch product lists
         rest.get('*/customers/:customerId/product-lists', (req, res, ctx) => {
             return res(ctx.json(mockedCustomerProductLists))
         }),
 
-        // mock adding guest email to basket
-        rest.put('*/baskets/:basketId/customer', (req, res, ctx) => {
-            currentBasket.customer_info.email = 'customer@test.com'
+        // mock add shipping and billing address to basket
+        rest.put('*/shipping-address', (req, res, ctx) => {
+            const shippingBillingAddress = {
+                address1: req.body.address1,
+                city: 'Tampa',
+                countryCode: 'US',
+                firstName: 'Test',
+                fullName: 'Test McTester',
+                id: '047b18d4aaaf4138f693a4b931',
+                lastName: 'McTester',
+                phone: '(727) 555-1234',
+                postalCode: '33712',
+                stateCode: 'FL'
+            }
+            currentBasket.shipments[0].shippingAddress = shippingBillingAddress
+            currentBasket.billingAddress = shippingBillingAddress
             return res(ctx.json(currentBasket))
         }),
 
-        // mock add shipping and billing address to basket
-        rest.put('*/shipping_address', (req, res, ctx) => {
+        // mock add billing address to basket
+        rest.put('*/billing-address', (req, res, ctx) => {
             const shippingBillingAddress = {
-                address1: req.body.address1,
+                address1: '123 Main St',
                 city: 'Tampa',
-                country_code: 'US',
-                first_name: 'Test',
-                full_name: 'Test McTester',
+                countryCode: 'US',
+                firstName: 'Test',
+                fullName: 'Test McTester',
                 id: '047b18d4aaaf4138f693a4b931',
-                last_name: 'McTester',
+                lastName: 'McTester',
                 phone: '(727) 555-1234',
-                postal_code: '33712',
-                state_code: 'FL',
-                _type: 'order_address'
+                postalCode: '33712',
+                stateCode: 'FL',
+                _type: 'orderAddress'
             }
-            currentBasket.shipments[0].shipping_address = shippingBillingAddress
-            currentBasket.billing_address = shippingBillingAddress
+            currentBasket.shipments[0].shippingAddress = shippingBillingAddress
+            currentBasket.billingAddress = shippingBillingAddress
+            return res(ctx.json(currentBasket))
+        }),
+
+        // mock add shipping method
+        rest.put('*/shipments/me/shipping-method', (req, res, ctx) => {
```

Comment:
Some of these mocks are repeated - once in `logInDuringCheckout()` and once in the test 'Can proceed through checkout steps as guest'.. Should we declare these only once in `beforeEach()`?

Reply:
The `shipping-method` mutation actually differs according to where it gets called. 

But the `shipping-methods` (plural) query is indeed repeated. Let me refactor it a bit.